makefile

```diff
@@ -27,40 +27,48 @@ setup-rust:
 aarch64-linux-android: CARGO_ENV_FLAGS += \
 	ISYSROOT=$(NDK_HOME)/toolchains/llvm/prebuilt/linux-x86_64/sysroot \
 	ISYSTEM=$(NDK_HOME)/toolchains/llvm/prebuilt/linux-x86_64/sysroot/usr/include/aarch64-linux-android \
+	AR=llvm-ar \
+	CFLAGS=-Wno-error=unused-but-set-parameter \
 	CXX=aarch64-linux-android$(MIN_API_LEVEL_64_BIT)-clang++ \
 	CC=aarch64-linux-android$(MIN_API_LEVEL_64_BIT)-clang \
 	CARGO_TARGET_AARCH64_LINUX_ANDROID_LINKER=aarch64-linux-android$(MIN_API_LEVEL_64_BIT)-clang \
-	CARGO_TARGET_AARCH64_LINUX_ANDROID_AR=aarch64-linux-android-ar \
+	CARGO_TARGET_DIR=target/aarch64 \
```

Comment:
TBH I'm not sure how this was working before.
It looks like all the different targets were using a common `target/build` for some of the outputs this resulted in incompatible code at link time.
I think this goes back to these kind of things in the MobileCoin repo
```toml
[dependencies]
mbedtls = { version = "0.8.1", default-features = false, features = ["no_std_deps"] }

[build-dependencies]
# We use mbedtls to generate certificates for simulation mode
mbedtls = "0.8.1"

```
The `dependencies` need to be built for the target, i.e. aarch android, but the build dependencies may need to be built for the host i.e. x86_64 linux.
Since mbedtls builds C files via CMake and above we set the CC to *always* be the android one. I think things start to collide.

Reply:
Thinking about this some more, I don't think the `dependencies` vs `build dependencies` is to blame here, since that behavior still happens when separating the build directories per target. I see there are multiple build directories per crate of the form `crate-#####` where the `#####` is a hash of some kind. It's more likely that we're not giving cargo enough information to disambiguate the different architectures, I would think the `--target <triple>` would do it...